specs/altair/sync-protocol.md

```diff
@@ -256,3 +351,395 @@ def process_light_client_update(store: LightClientStore,
+    process_light_client_update(
+        store,
+        LightClientUpdate(
+            attested_header=optimistic_update.attested_header,
+            next_sync_committee=SyncCommittee(),
+            next_sync_committee_branch=[Bytes32() for _ in range(floorlog2(NEXT_SYNC_COMMITTEE_INDEX))],
+            finalized_header=BeaconBlockHeader(),
+            finality_branch=[Bytes32() for _ in range(floorlog2(FINALIZED_ROOT_INDEX))],
+            sync_aggregate=optimistic_update.sync_aggregate,
+            fork_version=optimistic_update.fork_version),
+        current_slot,
+        genesis_validators_root)
+```
+
+## Networking
+
+This section extends the [networking specification for Altair](./p2p-interface.md) with additional messages, topics and data to the Req-Resp and Gossip domains.
+
+### Configuration
+
+| Name | Value | Description |
+| - | - | - |
+| `MAX_REQUEST_LIGHT_CLIENT_UPDATES` | `2**7` (= 128) | Maximum number of `LightClientUpdate` instances in a single request |
+
+### The gossip domain: gossipsub
+
+A gossip mesh is added to allow light clients to stay in sync with the network.
+
+#### Topics and messages
+
+A new global topic is added to provide light clients with the latest updates for
+staying in sync with the network.
+
+| name | Message Type |
+| - | - |
+| `optimistic_light_client_update` | `OptimisticLightClientUpdate` |
+
+##### Global topics
+
+###### `optimistic_light_client_update`
+
+This topic is used to propagate the latest `OptimisticLightClientUpdate` to light
+clients, allowing them to stay in sync with the network.
+
+The following validations MUST pass before forwarding the `optimistic_update`
+on the network.
+- _[IGNORE]_ The optimistic update is not attesting to the latest block's parent block --
+  i.e. validate that `hash_tree_root(optimistic_update.attested_header) == state.latest_block_header.parent_root`.
+- _[REJECT]_ The optimistic update has an unexpected fork version --
+  i.e. validate that `optimistic_update.fork_version == state.fork.current_version`
+- _[REJECT]_ The optimistic update sync aggregate differs from the one included in the latest block --
+  i.e. with `block` referring to the latest block,
+  validate that `optimistic_update.sync_aggregate == block.message.body.sync_aggregate`
+
+Note that the `sync_aggregate` has already been validated when processing the
+latest block. While different valid aggregate signatures may be formed,
+only a single canonical one (the same one that was included in the block) should
+be propagated across the network.
+
+### The Req/Resp domain
+
+#### Messages
+
+##### LightClientUpdatesByRange
+
+**Protocol ID:** `/eth2/beacon_chain/req/light_client_updates_by_range/1/`
+
+Request Content:
+```
+(
+  start_period: uint64
+  count: uint64
+  step: uint64
+)
+```
+
+Response Content:
+```
+(
+  List[LightClientUpdate, MAX_REQUEST_LIGHT_CLIENT_UPDATES]
+)
+```
+
+Requests `LightClientUpdate` instances in the sync committee period range
```

Comment:
This could lead a very large response.

Reply:
The response contains a series of chunks, each only having a single `LightClientUpdate`. There are at most as many chunks as have been requested.

This follows the same methodology as the existing `BeaconBlocksByRange` endpoint:
https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#beaconblocksbyrange